x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/SourceDestValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.ml.dataframe;
+
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.regex.Regex;
+import org.elasticsearch.xpack.core.ml.dataframe.DataFrameAnalyticsConfig;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+public class SourceDestValidator {
+
+    private final ClusterState clusterState;
+    private final IndexNameExpressionResolver indexNameExpressionResolver;
+
+    public SourceDestValidator(ClusterState clusterState, IndexNameExpressionResolver indexNameExpressionResolver) {
+        this.clusterState = Objects.requireNonNull(clusterState);
+        this.indexNameExpressionResolver = Objects.requireNonNull(indexNameExpressionResolver);
+    }
+
+    public void check(DataFrameAnalyticsConfig config) {
```

Comment:
I suppose it is just a Java thing, but I have trouble with creating an object with state, but an internal method is only called once. Seems like a static `check(ClusterState clusterState, IndexNameExpressionResolver indexNameExpressionResolver, DataFrameAnalyticsConfig config)` is fewer lines of code and less state to break. 

Of course, I am hypocritical with this :). I often make these classes myself.

Reply:
I know what you mean. I insist going for objects though because I think they lend themselves better for expansion. On the other hand, when one sees a static method it discourages refactoring to make an object. Of course, only time will tell. My view is that when paradigms are not clearly better or worse than alternatives, we have to try them out and wait for empirical evidence to reward us or slap us in the face :-)